compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -3144,7 +3144,23 @@ class JSCodeGen()(using genCtx: Context) {
       val tpe = atPhase(elimErasedValueTypePhase) {
         sym.info.finalResultType
       }
-      unbox(boxedResult, tpe)
+      if (tpe.isRef(defn.BoxedUnitClass) && sym.isGetter) {
+        /* Work around to reclaim Scala 2 erasure behavior, assumed by the test
+         * NonNativeJSTypeTest.defaultValuesForFields.
+         * Scala 2 erases getters of `Unit`-typed fields as returning `Unit`
+         * (not `BoxedUnit`). Therefore, when called in expression position,
+         * the call site introduces an explicit `BoxedUnit.UNIT`. Even if the
+         * field has not been initialized at all (with `= _`), this results in
+         * an actual `()` value.
+         * In Scala 3, the same pattern returns `null`, as a `BoxedUnit`, so we
```

Comment:
Could we instead change the behavior of Scala 3? We already have some logic for "unboxing" nulls from value classes: https://github.com/lampepfl/dotty/blob/3636ee6d0fda29daf9ef6e4e0491e2e504b227ed/compiler/src/dotty/tools/dotc/transform/Erasure.scala#L292-L317

Reply:
We cannot change it without breaking binary compatibility ;)